primefaces/src/main/java/org/primefaces/component/api/UICalendar.java

```diff
@@ -368,10 +369,9 @@ protected void createFacesMessageFromValidationResult(FacesContext context, Vali
         }
     }
 
-    @SuppressWarnings({"unchecked", "rawtypes"})
     public void validateMinMax(FacesContext context) {
-        Comparable minDate = (Comparable) getMindate();
-        Comparable maxDate = (Comparable) getMaxdate();
+        LocalDate minDate = CalendarUtils.getObjectAsLocalDate(context, this, getMindate());
+        LocalDate maxDate = CalendarUtils.getObjectAsLocalDate(context, this, getMaxdate());
```

Comment:
IMO this is a bit more complicated. mindate/maxdate may be java.time.LocalDate, java.time.LocalDateTime, java.time.LocalTime, java.util.Date (deprecated) or String when we look into docu.
Currently it is broken for String. But we should not break this for LocalDateTime what this PR does. (mindate may be something like '27.05.2021 10:00' for a LocalDateTime-usecase)
Maybe we should explicit check for String and only do some explizit conversion for String.

Reply:
Agreed.

Reply:
These both should equate to an `Instant` somehow.

Reply:
… or we may convert everything to `LocalDateTime`.

Reply:
I picked @melloware's suggestion to convert evrything to `Instant`
```
    private Instant toInstant(FacesContext context, Object date) {
        if (date == null) {
            return null;
        }
        if (date instanceof LocalDateTime) {
            return ((LocalDateTime) date).toInstant(ZoneOffset.UTC);
        }
        if (date instanceof LocalDate) {
            return ((LocalDate) date).atStartOfDay().toInstant(ZoneOffset.UTC);
        }
        if (date instanceof LocalTime) {
            LocalDate now = LocalDate.now();
            return now.atTime(((LocalTime) date)).toInstant(ZoneOffset.UTC);
        }
        if (date instanceof Date) {
            if (date instanceof java.sql.Date) {
                // java.sql.Date does not support toInstant
                return new Date(((java.sql.Date) date).getTime()).toInstant();
            }
            return ((Date) date).toInstant(); // implied UTC
        }
        if (date instanceof String) {
            return CalendarUtils.getObjectAsLocalDate(context, this, date).atStartOfDay().toInstant(ZoneOffset.UTC);
        }

        throw new FacesException(... " unsupported type ");
    }
```
@christophs78  I'm not sure if the `String` case is sufficient. Is it possible that the min/maxDate string may contain time parts?

Reply:
@christophs78 Looks like it is possible to set time parts in min/maxDate string. At least `DatePicker.validateDateValue` takes it into account. So the `String` case has to be improved.